src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,57 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include <yaml-cpp/yaml.h>
+#include "rgw_tracer.h"
+#include "common/debug.h"
+#include "global/global_context.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+thread_local std::unique_ptr<tracing::Tracer> rgw_tracer = std::unique_ptr<tracing::Tracer>(new tracing::Tracer(jaeger_configuration::jaeger_default_config));
+
+namespace tracing {
+
+const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+
+Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {};
+
+Tracer::Tracer(const std::string& config_file_path) {
+  const auto config_yaml = YAML::LoadFile(config_file_path);
+  const auto config = jaegertracing::Config::parse(config_yaml);
+  open_tracer =  jaegertracing::Tracer::make(config);
+};
+
+std::unique_ptr<opentracing::Span> Tracer::start_trace(const std::string& trace_name) {
+  if(is_enabled()) {
+    return open_tracer->StartSpan(trace_name);
+  }
+  return noop_tracer->StartSpan(trace_name);
+}
+
+std::unique_ptr<opentracing::Span> Tracer::start_span(const std::string& span_name, std::unique_ptr<opentracing::Span>& parent_span) {
```

Comment:
@zenomri @yuvalif can you help me understand why we are keeping/updating these to be rgw only and not in common https://github.com/ceph/ceph/blob/master/src/common/tracer.{cc/h}

Reply:
hey, the idea is to do it in the right way inside the RGW and then update the common